tests/cell_test.cc

```diff
@@ -489,6 +489,49 @@ TEST_CASE("Cell is checked for 2D case", "[cell][2D]") {
     }
   }
 
+  // Check material adddition to nodes
+  SECTION("Check material id addition to nodes") {
+    auto cell = std::make_shared<mpm::Cell<Dim>>(0, Nnodes, element);
+    // add nodes to cell
+    cell->add_node(0, node0);
+    cell->add_node(1, node1);
+    cell->add_node(2, node2);
+    cell->add_node(3, node3);
+
+    // add material ids to nodes in cell
+    cell->append_material_id_to_nodes(2);
+    cell->append_material_id_to_nodes(1);
+    cell->append_material_id_to_nodes(1);
+
+    // check amount of material ids added to each node
+    REQUIRE(node0->material_ids().size() == 2);
+    REQUIRE(node1->material_ids().size() == 2);
+    REQUIRE(node2->material_ids().size() == 2);
+    REQUIRE(node3->material_ids().size() == 2);
+
+    // check which material ids were added to each node
+    int id_check = 1;
+    for (auto id_itr : node0->material_ids()) {
```

Comment:
Check for the size, and maybe just set a vector to iterate over, this check is confusing